snowplow-tracker/src/main/java/com/snowplowanalytics/snowplow/internal/tracker/ServiceProvider.java

```diff
@@ -233,79 +232,95 @@ private void initializeConfigurationUpdates() {
     // Getters
 
     @NonNull
-    public Subject getSubject() {
+    @Override
+    public Subject getOrMakeSubject() {
         if (subject == null) {
             subject = makeSubject();
         }
         return subject;
     }
 
     @NonNull
-    public Emitter getEmitter() {
+    @Override
+    public Emitter getOrMakeEmitter() {
         if (emitter == null) {
             emitter = makeEmitter();
         }
         return emitter;
     }
 
     @NonNull
-    public Tracker getTracker() {
+    @Override
+    public Boolean isTrackerInitialized() {
+        return tracker != null;
+    }
```

Comment:
Same thought I already added on iOS Tracker.
Rethinking to this, maybe we could call this method "isAlive" or "isActive" or in negative form "isInvalidated", related to the fact that it's the ServiceProvider that should be dead but it's still used.
Just a thought btw.